File: 백준/Silver/1074. Z/Z.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    static int count = 0;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

        int N = Integer.parseInt(st.nextToken());
        int r = Integer.parseInt(st.nextToken());
        int c = Integer.parseInt(st.nextToken());

        int size = (int)Math.pow(2, N);
        recur(size, r, c);
        System.out.println(count);
    }

     static void recur (int size, int r, int c){
        if (size == 1){
            return;
        }
        else if (r < size/2 && c < size/2){
            recur(size/2, r, c);
        }
        else if (r < size/2 && c >= size/2){
            count += size*size/4;
            recur (size/2, r, c - size/2);
        }
        else if (r >= size/2 && c < size/2){
            count += (size*size/4) * 2;
            recur (size/2, r - size/2, c);
        }
        else {
            count += (size * size / 4) * 3;
            recur(size/2, r - size/2, c - size/2);
        }
    }
}

File: 백준/Silver/1074. Z/Z.c
#include <stdio.h>
#include<math.h>

int count = 0;

void recur(int N, int r, int c){
    if (N == 1){
        return ;
    }
    else if ((r < N/2) && (c < N/2)){
        recur(N/2, r, c);
    }
    else if ((r < N/2) && (c >= N/2)){
        count += N*N/4;
        recur (N/2, r, c - N/2);
    }
    else if (r >= N/2 && c < N/2){
        count += (N*N/4) * 2;
        recur (N/2, r - N/2, c);
    }
    else {
        count += (N * N / 4) * 3;
        recur(N/2, r - N/2, c - N/2);
    }
}

int main(){
    int N, r, c;
    scanf("%d %d %d", &N, &r, &c);

    int size = pow(2,N);
    recur(size, r, c);

    printf("%d", count);


    return 0;
}